lib/Parser/Parse.cpp

```diff
@@ -12692,19 +12729,52 @@ ParseNodeUni * Parser::ParseDestructuredObjectLiteral(tokens declarationType, bo
         declarationType = tkLCurly;
     }
     ParseNodePtr pnodeMemberList = ParseMemberList<buildAST>(nullptr/*pNameHint*/, nullptr/*pHintLength*/, declarationType);
+
+    // Count the number of non-rest members in the object
+    uint32 staticCount = 0;
+    uint32 computedCount = 0;
+    bool hasRest = false;
+    if (pnodeMemberList != nullptr)
+    {
+        Assert(pnodeMemberList->nop == knopList || pnodeMemberList->nop == knopObjectPatternMember || pnodeMemberList->nop == knopEllipsis);
+
+        ForEachItemInList(pnodeMemberList, [&](ParseNodePtr current) {
+            if (current->nop != knopEllipsis)
```

Comment:
should we refactor this one a function. The code duplicated from ConvertObjectToObjectPattern as well?

Reply:
agreed

---
In reply to: [205214793](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r205214793) [](ancestors = 205214793)